repose-aggregator/components/filters/uri-stripper-filter/src/test/scala/org/openrepose/filters/uristripper/UriStripperFilterTest.scala

```diff
@@ -114,6 +118,321 @@ class UriStripperFilterTest extends FunSpec with BeforeAndAfterEach with Matcher
+      val config =
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+           |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+           |    <link-resource uri-path-regex=".*">
+           |        <json link-mismatch-action="remove">$$.dne</json>
+           |    </link-resource>
+           |</uri-stripper>
+         """.stripMargin
+
+      val respBody =
+        s"""
+           |{
+           |  "link": "http://example.com/v1/foo"
+           |}
+         """.stripMargin
+
+      filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+      request.setRequestURI("/v1/12345/foo")
+      setResponseBody(respBody, MimeType.APPLICATION_JSON.toString)
+
+      filter.doFilter(request, response, filterChain)
+
+      response.getContentAsString shouldEqual Json.parse(respBody).toString()
+    }
+
+    it("should remove the field if the token index is out of bounds (remove)") {
+      val config =
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+            |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+            |    <link-resource uri-path-regex=".*">
+            |        <json link-mismatch-action="remove" token-index="5">$$.link</json>
+            |    </link-resource>
+            |</uri-stripper>
+         """.stripMargin
+
+      val respBody =
+        s"""
+           |{
+           |  "link": "http://example.com/v1/foo"
+           |}
+         """.stripMargin
+
+      filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+      request.setRequestURI("/v1/12345/foo")
+      setResponseBody(respBody, MimeType.APPLICATION_JSON.toString)
+
+      filter.doFilter(request, response, filterChain)
+
+      response.getContentAsString shouldEqual "{}"
+    }
+
+    it("should fail if the link cannot be located (fail)") {
+      val config =
+        s"""<?xml version="1.0" encoding="UTF-8"?>
+           |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+           |    <link-resource uri-path-regex=".*">
+           |        <json link-mismatch-action="fail">$$.dne</json>
+           |    </link-resource>
+           |</uri-stripper>
+         """.stripMargin
+
+      val respBody =
+        s"""
+           |{
+           |  "link": "http://example.com/v1/foo"
+           |}
+         """.stripMargin
+
+      filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+      request.setRequestURI("/v1/12345/foo")
+      setResponseBody(respBody, MimeType.APPLICATION_JSON.toString)
+
+      filter.doFilter(request, response, filterChain)
+
+      response.getStatus shouldEqual HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+      response.getContentLength shouldEqual 0
+    }
+
+    it("should fail if the link's token index is out of bounds (fail)") {
```

Comment:
You've tested indexing failures, but what about if they haven't specified the index, and it can't figure out where to place it based on pre and post token?